images/ci-prune-buildcache/ci_buildcache_prune.py

```diff
@@ -0,0 +1,497 @@
+import argparse
+import boto3
+from collections import defaultdict
+from contextlib import contextmanager
+from datetime import datetime, timedelta, timezone
+import gitlab
+import multiprocessing as mp
+import multiprocessing.pool as pool
+import os
+from io import BytesIO
+import sentry_sdk
+from urllib.parse import urlparse
+
+import buildcache
+import helper
+from pruner import DirectPruner, IndexPruner, OrphanPruner
+
+sentry_sdk.init(
+    # This cron job only runs once weekly,
+    # so just record all transactions.
+    traces_sample_rate=1.0,
+)
+
+# Script Config
+PRUNE_REF = os.environ.get("PRUNE_REF", "develop")
+SINCE_DAYS = int(os.environ.get("PRUNE_SINCE_DAYS", 14))
+
+# Gitlab API (for fetching artifacts)
+GITLAB_URL = os.environ.get("GITLAB_URL", "https://gitlab.spack.io")
+GITLAB_PROJECT = os.environ.get("GITLAB_PROJECT", "spack/spack")
+gl = gitlab.Gitlab(GITLAB_URL)
+
+# S3
+BUILDCACHE_URL = os.environ.get("BUILDCACHE_URL", f"s3://spack-binaries/{PRUNE_REF}").rstrip("/")
```

Comment:
Can this fail if it isn't in the environment? Otherwise I'd rather staging not default to trying to mess with the production bucket.

Reply:
This is setup to make the local running of the script less of a hassle. Credentials are required to actually perform the  destructive operation so even if this does run in staging incorrectly configured it shouldn't have credentials to modify production.